cpp/ql/src/semmle/code/cpp/PrintAST.qll

```diff
@@ -114,9 +114,32 @@ class PrintASTNode extends TPrintASTNode {
 
   /**
    * Gets the child node at index `childIndex`. Child indices must be unique,
-   * but need not be contiguous (but see `getChildByRank`).
+   * but need not be contiguous.
    */
-  abstract PrintASTNode getChild(int childIndex);
+  abstract PrintASTNode getChildInternal(int childIndex);
+
+  /**
+   * Gets the child node at index `childIndex`.
+   * Adds edges to fully converted expressions, that are not part of the
+   * regular parent/child relation traversal.
+   */
+  PrintASTNode getChild(int childIndex) {
```

Comment:
```suggestion
  final PrintASTNode getChild(int childIndex) {
```
Can this be final? The rest of this library is fairly consistent with using `final` where possible.